tests/test_env.py

```diff
@@ -231,19 +231,68 @@ def test_process_env_install_from_index(
 
 
 @pytest.mark.slowtest
+@pytest.mark.parametrize_any("extra_indexes", [[], ["http://example.com/extra_index_1", "http://example.com/extra_index_2"]])
 def test_process_env_install_from_index_not_found(
-    tmpvenv_active: tuple[py.path.local, py.path.local], local_module_package_index: str
+    tmpvenv_active: tuple[py.path.local, py.path.local],
+    local_module_package_index: str,
+    extra_indexes,
 ) -> None:
     """
     Attempt to install a package that does not exist from a pip index. Assert the appropriate error is raised.
     """
-    with pytest.raises(env.PackageNotFound, match="Packages this-package-does-not-exist were not found in the given indexes."):
+    indexes = local_module_package_index
+    if extra_indexes:
+        indexes = local_module_package_index + ", " + ", ".join(extra_indexes)
+
+    expected: str = (
+        "Packages this-package-does-not-exist were not found in the given indexes. " "(Looking in indexes: %s)" % indexes
+    )
+    with pytest.raises(env.PackageNotFound, match=re.escape(expected)):
         # pass use_system_config=False for security reasons (anyone could publish this package to PyPi)
         env.process_env.install_for_config(
             [Requirement.parse("this-package-does-not-exist")],
-            config=PipConfig(
-                index_url=local_module_package_index,
-            ),
+            config=PipConfig(index_url=local_module_package_index, extra_index_url=extra_indexes if extra_indexes else []),
+        )
```

Comment:
`extra_indexes` is always a list, right?
```suggestion
            config=PipConfig(index_url=local_module_package_index, extra_index_url=extra_indexes),
        )
```